docs/en/observability/synthetics-run-test.asciidoc

```diff
@@ -0,0 +1,146 @@
+[[synthetic-run-tests]]
+= Run synthetic tests
+
+++++
+<titleabbrev>Run synthetic tests</titleabbrev>
+++++
+
+There are two ways to run synthetic tests:
+
+* If you want to create a single journey and manage it individually, <<synthetics-inline-journey,use an inline monitor>>.
+Inline monitors can be more efficient, but have some limitations like not supporting dependencies or more than one journey.
+* If you have more than one journey, rely on dependencies, or your tests need to live with your
+application code, <<synthetics-test-suite,use a test suite>>.
+
+[discrete]
+[[synthetics-inline-journey]]
+== Use an inline monitor
+
+Inline monitors are quick and easy to set up.
+To create one, you can copy a journey's content, paste it into Kibana or your {heartbeat} configuration, and save.
```

Comment:
I'm wondering if we should put suites first.

Reading this, I see now that we have some serious UX work to do around inline monitors. So, there is a way to write inline monitors locally without using a full suite, but it's undocumented and confusing. The way to do that would be to put it in a file then invoke `cat myinline.ts | npx @elastic/synthetics --stdin`.

Now, I'm not saying we should actually document that, because that's terrible. The challenge with inline monitors is that they're really only useful for the case where we give you an example that you can copy/paste to get going. It's really hard to work on / iterate with them since, unlike suites, they don't benefit from type checking etc. that suites get being part of an NPM project.

In short, I think you wrote the right thing here, but it's something I'd like to chew on for ways to improve. My suspicion is that in the future we'll focus on a smoother script recorder flow and suites, and maybe deprecate inline monitors.

CC @paulb-elastic @drewpost @lucasfcosta @vigneshshanmugam

Reply:
That makes total sense. I do think putting suites first would be much better, as that's what we think is the best way of using the product. IMO it would also be worth highlighting that we recommend people to use suites rather than inline monitors.

However, I'd be concerned with deprecating inline monitors. When I put myself into a user's shoes, I'd probably write a couple of inline monitors to get started with the product more quickly, and see how it works before buying in and setting up a more complex infrastructure (that's a GH repo, or the necessary automations/hooks for running things in CI and locally).

For now I also think this is the right thing to do, and would quite like to have suites first especially once we improve the UX with the monitor management UI.

Reply:
My worry around people getting their feet wet with inline monitors is you don't have:

1. The easy ability to iterate and debug locally, so lot's of trial and error revolving around 'run now' in the UI
2. No syntax suggestions from typescript etc.

So, I think it's just as frustrating.

My hope is that with the push command local dev of scripts gets so easy inline largely doesn't make sense. I could be off though.